sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java

```diff
@@ -179,7 +181,7 @@ private void loadAzureVmMetaData() {
             return OBJECT_MAPPER.readValue(itemResponseBodyAsString, itemClassType);
         } catch (IOException e) {
             throw new IllegalStateException(
-                String.format("Failed to parse string [%s] to POJO.", itemResponseBodyAsString, e));
+                String.format("Failed to parse string [%s] to POJO.", itemResponseBodyAsString), e);
```

Comment:
Thanks for fixing this formatting call @FabianMeiswinkel . 
However, I strongly believe we can optimize here by not creating a `new Formatter()` object for every response, instead we should update to simple string concatenation for this exception message. 
```suggestion
                "Failed to parse string [" + itemResponseBodyAsString + "] to POJO.", e);
```